CONTRIBUTING.md

```diff
@@ -188,7 +188,63 @@ with one another are rolled up.
 Speaking of tests, Rust has a comprehensive test suite. More information about
 it can be found [here][rctd].
 
-### External Dependencies
+### External Dependencies (subtree)
+
+As a developer to this repository, you don't have to treat the following external projects
+differently from other crates that are directly in this repo:
+
+* none so far, see https://github.com/rust-lang/rust/issues/70651 for more info
+
+They are just regular files and directories. This is in contrast to `submodule` dependencies
+(see below for those). Only tool authors will actually use any operations here.
+
+#### Synchronizing a subtree
+
+There are two synchronization directions: `subtree push` and `subtree pull`.
+
+```
+git subtree push -P src/tools/clippy https://github.com/rust-lang/rust-clippy.git
+```
+
+takes all the changes that
+happened to the copy in this repo and creates commits on the remote repo that match the local
+changes (so every local commit that touched the subtree causes a commit on the remote repo).
+
+```
+git subtree pull -P src/tools/clippy https://github.com/rust-lang/rust-clippy.git
```

Comment:
With push and pull, also a `<ref>` has to be specified. There should be a note, that `push` should **never** aim at `master`, but at a new branch (either in a fork or the main repo) and to open a PR, so every sync is checked by the Clippy CI.

```suggestion
git subtree pull -P src/tools/clippy https://github.com/rust-lang/rust-clippy.git master
```

Reply:
Also, `.git` suffixes are never required, and it's cleaner to omit them, especially in docs.